xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java

```diff
@@ -137,6 +155,63 @@ protected List<SearchResult> search(List<SearchScope> searchScopes, String keywo
         }
     }
 
+    private List<SearchResult> searchPagesSolr(List<SearchScope> searchScopes, String keywords, String wikiName,
+        String space, int number, int start, String orderField, String order, Boolean withPrettyNames,
+        Boolean isLocaleAware) throws QueryException, XWikiException
+    {
+        if (StringUtils.isBlank(keywords)) {
+            return List.of();
+        }
+
+        List<String> filterQueries = new ArrayList<>();
+        filterQueries.add("type:DOCUMENT");
+        filterQueries.add("hidden:false");
+        if (StringUtils.isNotBlank(wikiName)) {
+            filterQueries.add("wiki:" + this.solrUtils.toCompleteFilterQueryString(wikiName));
+        }
+        if (StringUtils.isNotBlank(space)) {
+            filterQueries.add("space_exact:" + this.solrUtils.toCompleteFilterQueryString(space));
+        }
+
+        String escapedKeyWords = this.solrUtils.toFilterQueryString(keywords);
+        Set<SearchScope> supportedScopes = EnumSet.of(TITLE, NAME, CONTENT);
+        String queryString = searchScopes.stream()
+            .filter(supportedScopes::contains)
+            .map(scope -> switch (scope) {
+                // Consider matches in the title as way more important.
+                case TITLE -> "(title:" + escapedKeyWords + "*)^4";
+                // Prefer matching in the name over the spaces, but only if the name is not "WebHome".
+                case NAME -> "spaces:" + escapedKeyWords + "* OR (name:" + escapedKeyWords + "* -name_exact:WebHome)^2";
+                case CONTENT -> "doccontent:" + escapedKeyWords;
+                default -> throw new IllegalStateException("Unexpected value: " + scope);
+            })
+            .collect(Collectors.joining(" OR "));
+
+        if (queryString.isEmpty()) {
+            // No supported scope.
+            return List.of();
+        }
+
+        if (Boolean.TRUE.equals(isLocaleAware)) {
+            // TODO: this might not be the right way to filter by language.
+            Locale currentLocale = this.localizationContext.getCurrentLocale();
+            filterQueries.add("locale:(\"\" OR %s OR %s)".formatted(currentLocale, currentLocale.getLanguage()));
```

Comment:
That's not what we do in https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/XWiki/SuggestSolrMacros.xml#L108 . If you want to match the parent locale as well then you need to do as indicated on https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearchMacros.xml#L746  i.e. use the ``locales`` (plural). But since this is not enabled for the default search page, I think it's enough to do:

```suggestion
            filterQueries.add("locale:(\"\" OR %s)".formatted(currentLocale));
```